libraries/AP_NavEKF3/AP_NavEKF3.cpp

```diff
@@ -1358,9 +1358,11 @@ bool NavEKF3::setOriginLLH(const Location &loc)
     if (!core) {
         return false;
     }
-    if (sources.getPosXYSource() == AP_NavEKF_Source::SourceXY::GPS) {
-        // we don't allow setting of the EKF origin if using GPS to prevent
-        // accidental setting of EKF origin with invalid position or height
+    if ((sources.getPosXYSource() == AP_NavEKF_Source::SourceXY::GPS) || common_origin_valid) {
```

Comment:
my guess is that we should use "validOrigin" (the local boolean) instead of "common_origin_valid".

Reply:
The reason I was using `common_origin_valid` is that at this level in `class NavEKF3`  `validOrigin` isn't available. It is part of the core level `class NavEKF3_core`. I guess we could loop through each core and check `validOrigin` but getting the common_origin_valid correct and using that at this higher level seems like the right thing to me still?

I agree the placement of where the cores are insured to have the same origin is odd. When looking at it It took me a bit to figure out how the cores were each getting set to the new the common_origin.